admin_guide/pruning_resources.adoc

```diff
@@ -343,3 +344,152 @@ Such tags tend to have just one image in its history, which effectively prevents
 them from ever being pruned.
 xref:../dev_guide/managing_images.adoc#tag-naming[Learn more about _istag_
 naming.]
+
+[[hard-pruning-registry]]
+== Hard Pruning the Registry
+
+The OpenShift Container Registry can accumulate blobs that are not referenced by
+the {product-title} cluster's etcd. The basic xref:pruning-images[Pruning
+Images] procedure therefore is unable to operate on them. These are called
+_orphaned blobs_.
+
+Orphaned blobs can occur from the following scenarios:
+
+- Manually running the `oc delete image <sha256:image-id>` command only removes
+the image from etcd, but not from the registry's storage.
+- Pushing to the registry initiated by *docker* daemon failures causes some blobs
+to get uploaded but the image manifest (which is uploaded as the very last
+component) does not. All unique image blobs become orphans.
+- An image being refused because of quota suffers similarly to the previous
+example.
+- The standard image pruner may delete an image manifest but be interrupted before
+it deletes the related blobs.
+- There was a bug in the registry pruner which failed to remove blobs that were
+supposed to be removed. Image objects referencing them were removed and the
+blobs became orphans.
+// Find this BZ
+
+_Hard pruning_ the registry is a separate procedure from basic image pruning
+that allows you to remove orphaned blobs. You should hard prune if you are
+running out of storage space in your OpenShift Container Registry and believe
+you have orphaned blobs.
+
+This should be an infrequent operation and is necessary only when you have
+evidence that significant numbers of new orphans have been created. Otherwise,
+you can perform standard image pruning at regular intervals, for example, once a
+day (depending on your number of images being created).
+
+To hard prune orphaned blobs the registry:
+
+. *Log in* to xref:../cli_reference/get_started_cli.adoc#basic-setup-and-login[the CLI] as a user with an
+xref:../architecture/additional_concepts/authentication.adoc#oauth[access token].
+
+. *(Optional) Run a basic image prune.* This is an optional step which removes
+additional images that are no longer needed. The hard prune does not remove
+images on its own. It only takes care of blobs stored in the registry storage.
+Therefore it is a good idea run this just before the hard prune.
++
+See xref:pruning-images[Pruning Images] for steps.
+
+. *Switch to read-only mode.* The most important thing to note is that registry
+must be switched to the read-only mode. If not running in this mode, any pushes
+happening at the same time as the prune will either:
++
+--
+- fail and cause new orphans, or
+- succeed although the images will not be pullable (because some of the
+referenced blobs were deleted).
+--
++
+Pushes will not succeed until the registry is switched back to read-write mode.
+Therefore, the hard prune must be carefully scheduled.
++
+To switch the registry to read-only mode, run the following:
```

Comment:
Wouldn't it be worth adding an information that you need to wait for the re-deployment to happen. And in case of disabled triggers manually deploy the app, so that these new env values are picked up?